block/block.go

```diff
@@ -129,13 +129,13 @@ func (m *Manager) applyBlock(block *types.Block, commit *types.Commit, blockMeta
 	}
 
 	// check if the proposer needs to be changed
-	switchRole := m.Executor.UpdateProposerFromBlock(m.State, block)
+	switchRole := m.Executor.UpdateProposerFromBlock(m.State, m.Sequencers, block)
 
 	// save sequencers to store to be queried over RPC
 	batch := m.Store.NewBatch()
-	batch, err = m.Store.SaveSequencers(block.Header.Height, &m.State.Sequencers, batch)
+	batch, err = m.Store.SaveProposer(block.Header.Height, m.State.GetProposer(), batch)
```

Comment:
I think we need to save the proposer before `UpdateProposerFromBlock`
otherwise for block H, we store the next propser.
it will solve the proposer=nil comments